pkg/validator/validator.go

```diff
@@ -49,12 +48,24 @@ func New(client openapi.Client) (*Validator, error) {
 // It will return errors when there is an issue parsing the object, or if
 // it contains fields unknown to the schema, or if the schema was recursive.
 func (s *Validator) Parse(document []byte) (schema.GroupVersionKind, *unstructured.Unstructured, error) {
-	metadata := metav1.TypeMeta{}
-	if err := yaml.Unmarshal(document, &metadata); err != nil {
+	// Initialize an empty unstructured object
+	obj := &unstructured.Unstructured{}
+
+	// Parse the entire document into the unstructured object
+	if err := yaml.Unmarshal(document, obj); err != nil {
```

Comment:
Pre-existing, but this method of extracting GVK/name from obejct using sigs.k8s.io's `yaml.Unmarshal` is horribly inefficient and results in parsing the object multiple times. 

Not commenting for you to fix now, just noting

Reply:
I understand. let me fix the PR for now Ill create another PR to approve this